mkosi/distributions/__init__.py

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from pathlib import Path
+from typing import TYPE_CHECKING, List
+
+if TYPE_CHECKING:
+    from ..backend import MkosiState
+
+
+class DistributionInstaller:
+    @classmethod
+    def install(cls, state: "MkosiState") -> None:
```

Comment:
Probably, but not sure... I think this import had some other side effects, will look into it.

Reply:
Yeah, I think `from __future__ import annotations` is the way to go.

Reply:
I've had a look at this, and I don't think so. That's a PEP that's been punted down the line for the past two Python releases and this week's release says [that it might not be the future](https://docs.python.org/3.11/whatsnew/3.11.html#pep-563-may-not-be-the-future), so we shouldn't depend on something that might be different than what might be the default in the future.

Reply:
Sure, it might not be the future, but whatever solution they come up with will likely not be all that different in usage, so I say we use `from __future__ import annotations` for now and deal with that problem later

Reply:
Ah, since there was no reaction anymore I thought this one has runs it's course, sorry.

I disagree. I don't think we should `from __future__ import annotation`. This feature is opt and still very much in flux  or under consideration upstream and has fallen so much out of favour that they've added the above-linked note in the release notes after punting on accepting the PEP for two releases. Opting into this, I think we would rely on an unstable feature for, what I think is, mostly aesthetic gain. I think it's saver to stay with the language defaults and wait this out.

Reply:
I'm fine with discussing that in a separate PR, but we already use `from __future__ import annotations` in our other code. So I'd still prefer that in this PR, we stick to that and then in another PR you can remove all uses of this and we can discuss it there.

But we shouldn't do this switch in the middle of another large PR, let's keep this one focused on the split

Reply:
Agreed, opened #1280 to discuss dropping the rest, which I had actually missed. The diff is rather small.